src/lib/components/view.svelte

```diff
@@ -24,19 +39,43 @@
 					}
 					outOfSync = false;
 					manualUpdate = true;
-					if (code === state.code && config === state.mermaid) {
-						// Do not render if there is no change in Code/Config
+					if (code === state.code && config === state.mermaid && panZoomEnabled === state.panZoom) {
+						// Do not render if there is no change in Code/Config/PanZoom
 						return;
 					}
 					code = state.code;
 					config = state.mermaid;
+					panZoomEnabled = state.panZoom;
 					const scroll = view.parentElement.scrollTop;
 					delete container.dataset.processed;
 					mermaid.initialize(Object.assign({}, JSON.parse(state.mermaid)));
 					mermaid.render('graph-div', code, (svgCode) => {
 						if (svgCode.length > 0) {
-							console.log(svgCode);
+							if (state.panZoom) {
```

Comment:
This function seems quite long. Would it be possible to extract it ?

Reply:
Done!